media_softlet/linux/common/codec/ddi/dec/ddi_decode_functions.cpp

```diff
@@ -1076,10 +1081,16 @@ VAStatus DdiDecodeFunctions::StatusReport(
                             mediaSurfaceHeapElmt->pSurface->curStatusReport.decode.errMbNum = (uint32_t)tempNewReport.numMbsAffected;
                             mediaSurfaceHeapElmt->pSurface->curStatusReport.decode.crcValue = (uint32_t)tempNewReport.frameCrc;
                             mediaSurfaceHeapElmt->pSurface->curStatusReportQueryState = DDI_MEDIA_STATUS_REPORT_QUERY_STATE_COMPLETED;
+                            break_loop = true;
                             break;
                         }
                     }
 
+                    // surface->curStatusReportQueryState changed, break out the loop.
+                    if (break_loop) {
```

Comment:
remove break_loop, just use if(j <mediaCtx->pSurfaceHeap->uiAllocatedHeapElements) or else of  1094?